src/provideInlineCompletionItems.ts

```diff
@@ -26,12 +27,16 @@ export default async function provideInlineCompletionItems(
     ) {
       return new vscode.InlineCompletionList([]);
     }
+
     const completionInfo = context.selectedCompletionInfo;
     if (completionInfo) {
       return await getLookAheadSuggestion(document, completionInfo, position);
     }
-
-    return await getInlineCompletionItems(document, position);
+    const completions = await getInlineCompletionItems(document, position);
+    if (completions.items.length > 0 && !isEmptyCompletion(completions.items[0])) {
+      await handleFirstSuggestionDecoration(position);
+    }
+    return completions;
```

Comment:
I'd move the `if` to inside the `handle`, so that the caller code (i.e this code) will not have to be aware of the business logic of the "first suggestion decoration" feature:

```suggestion
const completions = await getInlineCompletionItems(document, position);
await handleFirstSuggestionDecoration(completion, position);
return completions;
```